Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "EditorGradientBakerComponent.h"
+
+#include <GradientSignal/Ebuses/GradientPreviewRequestBus.h>
+
+
+namespace GradientSignal
+{
+    void GradientBakerConfig::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serialize)
+        {
+            serialize->Class<GradientBakerConfig, AZ::ComponentConfig>()
+                ->Version(1)
+                ->Field("Gradient", &GradientBakerConfig::m_gradientSampler)
+                ->Field("InputBounds", &GradientBakerConfig::m_inputBounds)
+                ;
+
+            AZ::EditContext* edit = serialize->GetEditContext();
+            if (edit)
+            {
+                edit->Class<GradientBakerConfig>(
+                    "Gradient Baker", "")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::DisplayOrder, 50) // There's no special meaning to 50, we just need this class to move down and display below any children
+                    ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::ShowChildrenOnly)
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+                    ->DataElement(0, &GradientBakerConfig::m_gradientSampler, "Gradient", "Input gradient to bake the output image from.")
+                    ->DataElement(0, &GradientBakerConfig::m_inputBounds, "Input Bounds", "Input bounds for where to sample the data.")
+                    ;
+            }
+        }
+    }
+
+    void EditorGradientBakerComponent::Reflect(AZ::ReflectContext* context)
+    {
+        GradientBakerConfig::Reflect(context);
+
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<EditorGradientBakerComponent, EditorComponentBase>()
+                ->Version(0)
+                ->Field("Configuration", &EditorGradientBakerComponent::m_configuration)
+                ->Field("PreviewEntity", &EditorGradientBakerComponent::m_previewEntityId)
+                ->Field("PreviewPosition", &EditorGradientBakerComponent::m_previewPosition)
+                ->Field("PreviewSize", &EditorGradientBakerComponent::m_previewSize)
+                ->Field("ConstrainToShape", &EditorGradientBakerComponent::m_constrainToShape)
```

Comment:
I don't think we need any of these fields. Since this is producing a finite-sized image that isn't mapped to world space anywhere, the various controls for changing the view of the output preview don't really make sense here. We should always preview exactly what's going to end up in the baked image.

Reply:
Thanks yeah I that's what I thought too but wanted to confirm with you. Will remove!